src/app/ad-authoring/ad-authoring.component.ts

```diff
@@ -42,7 +42,9 @@ export class AdAuthoringComponent {
   constructor(private service: AdAuthoringService) {}
 
   updateLandingUrl(landingUrl: string) {
-    this.service.updateLandingUrl(landingUrl);
+    !this.urlControl.invalid
+      ? this.service.updateLandingUrl(landingUrl)
+      : this.service.updateLandingUrl('');
```

Comment:
We can use the ternary to determine just the URL instead of branching to separate service calls.

```suggestion
   this.service.updateLandingUrl(this.urlControl.invalid ? '' : landingUrl);
```

Reply:
completed in final touches PR